hack/e2e/resources/parallel-upgrade-checker-job.yaml

```diff
@@ -0,0 +1,52 @@
+kind: Job
+apiVersion: batch/v1
+metadata:
+  name: parallel-upgrades-checker
+  namespace: wmco-test
+  labels:
+    batch.kubernetes.io/job-name: parallel-upgrades-checker
+    job-name: parallel-upgrades-checker
+spec:
+  parallelism: 1
+  completions: 1
+  backoffLimit: 3
+  template:
+    metadata:
+      labels:
+        batch.kubernetes.io/job-name: parallel-upgrades-checker
+        job-name: parallel-upgrades-checker
+    spec:
+      nodeSelector:
+        kubernetes.io/os: linux
+      restartPolicy: Never
+      serviceAccountName: wmco-test
+      os:
+        name: linux
+      containers:
+        - name: parallel-upgrades-checker
+          image: 'REPLACE_WITH_OPENSHIFT_TOOLS_IMAGE'
+          command:
+            - bash
+            - '-c'
+            - |
+              #!/bin/bash
+              set -euo pipefail
+
+              # max number of parallel upgrades allowed, fixed to 1. Refer to controllers.MaxParallelUpgrades
+              export MAX_PARALLEL_UPGRADES=1
+
+              # loop indefinitely until count exceeded
+              while true; do
+              	upgradingCount=$(oc get nodes -l kubernetes.io/os=windows  -o jsonpath='{.items[*].metadata.labels.windowsmachineconfig\.openshift\.io/upgrading}' | wc -w)	
+              	if [[ $upgradingCount -gt $MAX_PARALLEL_UPGRADES ]]; then
+              	  echo "error: max upgrading count exceeded"
+              	  exit 1 
+              	fi
+
+              	echo "pass: upgrading count $upgradingCount/$MAX_PARALLEL_UPGRADES"
+              	echo "waiting 30s for next check..."
+              	sleep 30
```

Comment:
30 seconds is a big window. Its unlikely, but possible to miss an upgrade here. Especially if WMCO becomes more efficient in the future. How about 5 seconds?

Reply:
i lean towards more, maybe 10 or 15 seconds. 5 seconds seems very quick for upgrade polling, especially since there is no risk of missing something i.e. no parallel upgrades

Reply:
How is there no risk of missing something?
Even if we don't miss a whole upgrade, we need to test there is no overlap.

Reply:
Its better to be safer than not. There is no penalty I can think of to having a few dozen extra checks.

Reply:
Addressed in [a2a6f6b](https://github.com/openshift/windows-machine-config-operator/commit/a2a6f6b28ae61e28f226a62d5501d1e8c26acdea)